### ISSUE TITLE
Create a content function

### ISSUE THREAD

**Issue by solleer** [Level-2/Transphporm#178]
@TomBZombie I saw your recent amendment to the wiki and was wondering if we will add the content function you use in the example.

**Comment by TomBZombie** [comment]
Yes, that was going to be my next addition but I was also thinking about other ways we could handle the if statement.

The alternative would be attributes:

```xml
<div visible="data(foo)=bar">
```

But I don't think that's quite so clear. 

Or `content-mode: unwrap` which replaces the  elements with its contents, but again, I don't think it's very clear. 

In this example, we'd also need to support if/else which again poses a slight issue:

```xml
<if data="loggedin" equals="true">
    <then>
         You are logged in
     </then>
    <else>
      Sorry, you must log in to view this page
    </else>
</if>
```

But we could make the `content()` function take a selector:

```php

if[equals][data(attr(data))=attr(value)] { content: content('then'); }
if[equals][data(attr(data))!=attr(value)] { content: content('else'); }

```

it would be up to the individual implementer to decide on the the exact syntax but this would open up some possibilities.

**Comment by garrettw** [comment]
This whole if/else bit makes me wonder if Transphporm is planning to depart from its original vision of having no code in templates.

**Comment by TomBZombie** [comment]
> This whole if/else bit makes me wonder if Transphporm is planning to depart from its original vision of having no code in templates.

Absolutely, the point I wanted to make is that you can design your own traditional template engine using transphporm as a backend. At the moment Transphporm can already do 95% of it so adding a couple of extra features to nudge it up to 100% isn't really a lot of effort.

Really, it's just a way to show off some of the features of the language in a way that people coming from smarty/twig/whatever will easily grasp

**Comment by TomBZombie** [comment]
weirdly that entire wiki page is vanished for me and i can't even view the revision history
